azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -1446,13 +1292,12 @@ func flattenKubernetesClusterWindowsProfile(profile *containerservice.ManagedClu
 	}
 }
 
-func expandKubernetesClusterNetworkProfile(d *schema.ResourceData) *containerservice.NetworkProfileType {
-	configs := d.Get("network_profile").([]interface{})
-	if len(configs) == 0 {
+func expandKubernetesClusterNetworkProfile(input []interface{}) *containerservice.NetworkProfileType {
+	if len(input) == 0 {
 		return nil
 	}
 
-	config := configs[0].(map[string]interface{})
+	config := input[0].(map[string]interface{})
```

Comment:
Nil check here.

Reply:
this is optional/computed - and the flatten handles making this an empty slice if nothing's configured - so this should be fine